darts/dataprocessing/transformers/window_transformer.py

```diff
@@ -0,0 +1,691 @@
+                                )
+                            )
+                        ),
+                        f"`window_transformation` at index {idx} must contain a positive integer or 0 "
+                        f"for the 'comp_id', or a non-empty list containing positive integers/0. ",
+                    )
+                    if isinstance(transformation["comp_id"], int):
+                        window_transformations[idx]["comp_id"] = [
+                            transformation["comp_id"]
+                        ]  # make list
+
+        self.window_transformations = window_transformations
+
+    def _transform_iterator(
+        self, series: Union[TimeSeries, Sequence[TimeSeries]]
+    ) -> Iterator[Tuple[TimeSeries, dict]]:
+
+        series = series2seq(series)
+        builtins = self.BUILTIN_TRANSFORMS
+
+        # check if series_id and comp_id exist in the series list
+        # run through the transformations
+        series_subset = []
+        for idx, transformation in enumerate(self.window_transformations):
+            # check whether different transformations are to be performed for different series and different components
+            if "series_id" not in transformation and "comp_id" not in transformation:
+                # apply the transformation to all series and all components
+                series_subset += [
+                    (s.univariate_component(c), transformation, builtins)
+                    for s in series
+                    for c in range(s.width)
+                ]
+
+            elif "series_id" in transformation and "comp_id" not in transformation:
+                # apply the transformation to a specific series and all its components
+                raise_if_not(
+                    len(series) - 1 >= max(transformation["series_id"]),
+                    f"`window_transformation` at index {idx} has a 'series_id' that is greater than "
+                    f"the number of series. ",
+                )
+                series_subset += [
+                    (series[s_idx].univariate_component(c), transformation, builtins)
+                    for s_idx in transformation["series_id"]
+                    for c in range(series[s_idx].width)
+                ]
+
+            elif "series_id" not in transformation and "comp_id" in transformation:
+                # apply the transformation to all series on a specific component only in each series
+                # test that component exists in each relevant series
+                raise_if_not(
+                    all(
+                        c_idx <= series[s_idx].width - 1
+                        for s_idx in range(len(series))
+                        for c_idx in transformation["comp_id"]
+                    ),
+                    "Some components are not available in the provided series.",
+                )
+
+                series_subset += [
+                    (s.univariate_component(c), transformation, builtins)
```

Comment:
supporting component names instead of indexes, you could do something like `s[c].values()`

Reply:
with the new version, supporting component names, we don't need to extract values, I pass the whole series, all the transformation are then managed in TimeSeries.window_transform()